module/zfs/vdev_label.c

```diff
@@ -934,19 +941,22 @@ vdev_label_init(vdev_t *vd, uint64_t crtxg, vdev_labeltype_t reason)
 	error = nvlist_pack(label, &buf, &buflen, NV_ENCODE_XDR, KM_SLEEP);
 	if (error != 0) {
 		nvlist_free(label);
+		abd_return_buf_copy(vp_abd, vp, sizeof (vdev_phys_t));
```

Comment:
This could be `abd_return_buf()`.  There's no reason to copy the borrowed buffer data back in to the abd if we're just going to immediately free it.